src/EditorFeatures/Core/Formatting/FormatCommandHandler.FormatSelection.cs

```diff
@@ -40,31 +43,64 @@ private bool TryExecuteCommand(FormatSelectionCommandArgs args, CommandExecution
                 return false;
             }
 
-            using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Formatting_currently_selected_text))
+            // we only support single selection for now
+            var selection = args.TextView.Selection.GetSnapshotSpansOnBuffer(args.SubjectBuffer);
+            if (selection.Count != 1)
             {
-                var buffer = args.SubjectBuffer;
+                return false;
+            }
+
+            var token = _listener.BeginAsyncOperation(nameof(ExecuteCommand));
+            _ = ExecuteCommandAsync(args, document, selection.Single()).CompletesAsyncOperation(token);
```

Comment:
so we were previously fully synchronous here right?  I think we should keep that.  Moving to async here just gives us the opportunity to have to block on the BG.

Reply:
use BWI but still block UI synchronously?

Reply:
well, if we have no async work, don't do anything.  just process the entire command synchronously.

If we do have async work (and let me know if we do), then only have the BWI for that codepath.

Reply:
I guess this is the question: will we ever have async work for formatting? AFAIK this is all synchronous now. This doesn't require a semantic model, so we will never do async work? @tmat is that right?

Reply:
it shoudl be a goal of ours that for C#/VB this stays synchronous forever.  If we are calling out to TS though, that will likely stay async forever as well.  This is a case where we would likely have a split in the handler for these two types of languages.

Reply:
> @tmat is that right?

Yes, that is right.

> This is a case where we would likely have a split in the handler for these two types of languages.

+1

Reply:
Can we move this change to TS? That is, our general language-agnostic interface stays synchronous -- like CommandHandler.ExecuteCommand is but then TS uses BWI to make it better experience for async impl.

Reply:
Sure, I'm happy to. Where is the TS specific code for this? Is it in our repo still?

Reply:
You can make the change in `VSTypeScriptFormattingInteractionService` (roslyn repo). Make all methods of `IFormattingInteractionService` synchronous and then use BWI in `VSTypeScriptFormattingInteractionService`